doc/security/cryptolib/cryptolib_api.md

```diff
@@ -71,29 +71,32 @@ Word buffers can be safely interpreted as byte streams by the caller; the bytes
 ### Key data structures
 
 Keys receive extra protection from the cryptolib.
-Public keys are represented in plain, "unblinded" form, but have a checksum to protect their integrity.
+Public keys are represented in plain, "unblinded" form, but include a checksum to protect them against accidental corruption.
 The checksum is implementation-specific and may change over time.
-Therefore, the caller should not compute the checksum themselves; use the key import/export functions to construct unblinded keys.
+The caller should use algorithm-specific routines to construct unblinded keys; see e.g. the ECC and RSA sections for details.
```

Comment:
It is worth documenting how a required update to the format to mitigate a vulnerability may look like in the field. Since the key blob is tied to `kCryptoLibVersion1`, does this mean that we would have to support key blob migration functions?

Reply:
Potentially, yes. An example of a format change between library versions might be using a different checksum function or more shares in the keyblob (e.g. 3 instead of 2). The version information should alert the library if it has received a key that is using an older formatting, and the idea is that, depending on the severity of the vulnerability we were trying to mitigate and the security configuration of the key, we should be able to either return an error code or simply perform the operation with the old-format key.

To migrate exportable keys, the caller can always export them and then re-import them. Hardware-backed keys can be re-created with a newer cryptolib version by simply passing the same diversification data to the `otcrypto_hw_backed_key` function. The only tough one is non-hardware-backed keys that are marked as "not exportable" in their configuration; we might need to provide a migration function for those. Although I'm wondering whether this is actually a useful category anyway, given that the caller is trusted to handle the unencrypted key shares and could just infer the blinding scheme to export these anyway. I think originally the justification was to prevent accidental export of keys that weren't intended to leave the cryptolib.